### PR TITLE
ENH: Function Operations

### PR DESCRIPTION
**Pull request by giovaniceotto**
## Pull request type

Please check the type of change your PR introduces:

- [x] Code base additions (bugfix, features)
- [ ] Code maintenance (refactoring, formatting, renaming, tests)
- [ ] ReadMe, Docs and GitHub maintenance
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements, depending on the type of PR:

- Code base additions (for bug fixes / features):

  - [ ] Tests for the changes have been added
  - [ ] Docs have been reviewed and added / updated if needed
  - [ ] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [ ] All tests (`pytest --runslow`) have passed locally

## What is the current behavior?

Enter text here...

## What is the new behavior?

Enter text here...

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

Enter text here...

### CODE REVIEW

**Comment by giovaniceotto** [comment]
- Issues with the integral method, which were focused on exact integration of spline interpolated Functions, are now solved.
  - https://github.com/RocketPy-Team/RocketPy/pull/334/commits/e5eefaecd25670d77feee595c04fc3f6aa6f40cd
  - https://github.com/RocketPy-Team/RocketPy/pull/334/commits/217741ee2b602fcf6d251224c6559096388cea4e
  - https://github.com/RocketPy-Team/RocketPy/pull/334/commits/4f1181623f3e7d869d24f271949b65cbd5609c85
- Thorough testing has been added for the `Function.integral` method.
  - https://github.com/RocketPy-Team/RocketPy/pull/334/commits/e7c2d98c0af89ef61d96692e96474b3bbe74e576
  - https://github.com/RocketPy-Team/RocketPy/pull/334/commits/2024ab5a046387c8038a50953febdcb09c49820b

**Review by MateusStano — CHANGES_REQUESTED** [review]
I think all is good now!

Just one question:

https://github.com/RocketPy-Team/RocketPy/blob/e5eefaecd25670d77feee595c04fc3f6aa6f40cd/rocketpy/Function.py#L2766-L2769

What is going on here?

**Comment by MateusStano** [comment]
Also, I found this in the `enh/liquid-motors` branch:

https://github.com/RocketPy-Team/RocketPy/blob/2381a6d6ababcf1c60cbe521eeac53db0ee52eaf/rocketpy/Function.py#L2282-L2304

Should this be added here too?

File: rocketpy/Function.py [hunk 1]
```diff
@@ -2270,15 +2322,326 @@ def integral(self, a, b, numerical=False):
...
+            source is a callable instead of a list of points.
+        upper : float
+            Upper limit of the new domain. Only required if the Function's
+            source is a callable instead of a list of points.
+
+        Returns
+        -------
+        result : Function
+            A Function whose domain and range have been inverted.
+        """
```
Comment (MateusStano) [inline rocketpy/Function.py:2434]: docs here are wrong

File: rocketpy/Function.py [hunk 2]
```diff
@@ -2270,15 +2322,326 @@ def integral(self, a, b, numerical=False):
...
+            )
+        else:
+            lower = 0 if lower is None else lower
+            return Function(
+                lambda x: self.integral(lower, x),
+                inputs=self.__inputs__,
+                outputs=[o + " Integral" for o in self.__outputs__],
+            )
+
+    def inverseFunction(self, approxFunc=None, tol=1e-4):
```
Comment (MateusStano) [inline rocketpy/Function.py:2414]: docs here are wrong

File: rocketpy/Function.py (hunk 1 above)
Comment (MateusStano) [inline rocketpy/Function.py:2434]: Docs here are wrong

File: rocketpy/Function.py [hunk 3]
```diff
@@ -2270,15 +2322,326 @@ def integral(self, a, b, numerical=False):
...
+        Parameters
+        ----------
+        func : Function
+            The function to be inputed into the function.
+
+        Returns
+        -------
+        result : Function
+            The result of inputing the function into the function.
+        """
```
Comment (MateusStano) [inline rocketpy/Function.py:2545]: Missing some docs here

**Comment by giovaniceotto** [comment]
> I think all is good now!
> 
> Just one question:
> 
> https://github.com/RocketPy-Team/RocketPy/blob/e5eefaecd25670d77feee595c04fc3f6aa6f40cd/rocketpy/Function.py#L2766-L2769
> 
> What is going on here?
> 

This is doctest! It's there so that we can verify if all examples in the docstrings are working as expected.

More info here: https://docs.python.org/3/library/doctest.html

File: rocketpy/Function.py (hunk 3 above)
Comment (phmbressan) [reply rocketpy/Function.py]: Should be fixed by 7847e45bdc4f09e40b1e5e13a1d2095dd6466f04.

File: rocketpy/Function.py (hunk 1 above)
Comment (phmbressan) [reply rocketpy/Function.py]: Should be fixed by d5c4eab967f2fab6416b9e327439f05d13482701.

**Review by MateusStano — APPROVED** [review]

**Pull request merged**